core/templates/dev/head/pages/collection_player/CollectionPlayer.js

```diff
@@ -108,6 +129,88 @@ oppia.controller('CollectionPlayer', [
         explorationId).getExplorationSummaryObject();
     };
 
+    // Returns the SVG parameters required to draw the curved path.
+    $scope.generatePathParameters = function() {
+      // The pathSvgParameters represents the final string of SVG parameters
+      // for the bezier curve to be generated. The default parameters represent
+      // the first curve ie. lesson 1 to lesson 3.
+      $scope.pathSvgParameters = 'M250 80  C 470 100, 470 280, 250 300';
+      var collectionNodeCount = $scope.collection.getCollectionNodeCount();
+      // The sParameterExtension represents the co-ordinates following the 'S'
+      // (smooth curve to) command in SVG.
+      var sParameterExtension = '';
+      $scope.svgHeight = $scope.MIN_HEIGHT_FOR_PATH_SVG_PX;
+      $scope.pathIconParameters = $scope.generatePathIconParameters();
+      if (collectionNodeCount === 1) {
+        $scope.pathSvgParameters = '';
+        return $scope.pathSvgParameters;
```

Comment:
See guidance here on ng-init: https://docs.angularjs.org/api/ng/directive/ngInit. I think we're abusing it.

Can you do two things: don't return anything in this function, and call this function once in the controller itself (i.e. call $scope.generatePathParameters()) to initialize $scope.pathSvgParameters? In addition, don't return early anywhere; just set $scope.pathSvgParameters to what it is supposed to be based on the various conditions. I think that would be cleaner.